File: src/codes/scattered-instaniation/php/ksrc.php
<?php

class Mailer {
  public function send(string $to, string $message): void {
    echo "メール送信: {$to} => {$message}\n";
  }
}

class Notifier {
  public function notify(string $userId, string $content): void {
    $mailer = new Mailer();
    $formatted = "[通知] {$content}";
    $mailer->send("{$userId}@example.com", $formatted);
  }
}

File: src/codes/scattered-instaniation/py/ksrc.py
class Mailer:
    def send(self, to: str, message: str):
        print(f"メール送信: {to} => {message}")

class Notifier:
    def notify(self, user_id: str, content: str):
        mailer = Mailer()
        formatted = f"[通知] {content}"
        mailer.send(f"{user_id}@example.com", formatted)
